src/main/java/com/bezro/shopRESTfulAPI/services/impl/CartServiceImpl.java

```diff
@@ -24,29 +24,47 @@ public class CartServiceImpl implements CartService {
     private final ProductService productService;
     private final UserService userService;
 
-    public CartItemResponse addCartItem(CreateCartItemDto cartItemDto, String username) {
-        //TODO: optimize requests in addCartItem and updateCartItem
+    public CartItemResponse addOrUpdateCartItem(CreateCartItemDto cartItemDto, String username, Long cartItemId) {
```

Comment:
addOrUpdate is doing two things. Perhaps, you will quickly realize its side-effect when handle cartItemId I will suggest to separate it into two method and then create a shared private method if you have shared code among them. Also, can be tested separately.

Reply:
Agree, don't know how I went this wrong way)

Reply:
refactored back)